dune/iga/nurbstrimfunctionality.hh

```diff
@@ -0,0 +1,624 @@
+//
+// Created by Henri on 14.03.2023.
+//
+
+#pragma once
+
+namespace Dune::IGA::Impl::Trim {
+
+  // The following numbering is used for edges and nodes:
+  /*
+   * 3 -- 2 -- 2
+   * |         |
+   * 3         1
+   * |         |
+   * 0 -- 0 -- 1
+   */
+
+  constexpr int clipperPrecision = 8;
+  constexpr int pathSamples      = 200;
+  constexpr double tolerance     = 1e-8;
+  // constexpr double epsilonPrecision = double(16) * std::numeric_limits<double>::epsilon();
+
+  using ctype        = double;
+  using ClipperPoint = Clipper2Lib::Point<ctype>;
+  using Point        = FieldVector<ctype, 2>;
+  using PointVector  = std::vector<Point>;
+
+  using CurveGeometry = NURBSPatchGeometry<1, 2>;
+
+  // To keep track of which point belongs to which edge or node we will use a pointMap
+  struct IntersectionPoint {
+    Point point;
+    bool alreadyVisited = false;
+
+    explicit IntersectionPoint(const ClipperPoint& _clipperPoint) : point({_clipperPoint.x, _clipperPoint.y}){};
+  };
+
+  using IntersectionPointMap    = std::map<int, std::vector<IntersectionPoint>>;
+  using IntersectionPointMapPtr = std::shared_ptr<std::map<int, std::vector<IntersectionPoint>>>;
```

Comment:
You should not define that just use `std::shared_ptr<IntersectionPointMap>` then the object you have at hand is clearer